### PR TITLE
Fix auto vertical mode

### PR DESCRIPTION
**Pull request by tsroten**
## Description

The auto vertical mode was using unformatted output (with its best guess of table delimiters) to calculate whether or not the output was too big. This makes it use the first line of the formatted output.

It's not fool-proof because of TSV and CSV formats where the first line (header), might be shorter than other lines, but the previous method of measuring width had the same problem. So, this improves in some but not all ways.

Also, adds behave tests for auto vertical mode.

## Checklist

- [x] I've added this contribution to the `changelog.md`.
- [x] I've added my name to the `AUTHORS` file (or it's already there).

### CODE REVIEW

**Comment by tsroten** [comment]
@meeuw I think this will clash with the generator pull request. What do you think?

File: mycli/main.py [hunk 1]
```diff
@@ -793,9 +793,10 @@ def format_output(self, title, cur, headers, expanded=False,
             rows = list(cur)
             formatted = self.formatter.format_output(
                 rows, headers, format_name='vertical' if expanded else None)
+            first_line = formatted[:formatted.find('\n')]
```
Comment (j-bennet) [inline mycli/main.py]: I guess splitting on `\n` and finding a longest line would be a bad idea because of performance?

**Comment by meeuw** [comment]
yes, this doesn't work with the generators, I've also noticed that the rows generator is unrolled to a list (I'll fix that in the output generator branch).

maybe we should move the auto vertical mode to the cli_helpers tabular_output?

**Comment by tsroten** [comment]
@meeuw I don't think CLI Helpers should know about terminal sizes. Plus, we'd have to add postprocessors (instead of preprocessors) in CLI Helpers to make it work (since we have to compare the formatted output). Which also causes problems for it being a generator.

File: mycli/main.py (hunk 1 above)
Comment (tsroten) [inline mycli/main.py]: @j-bennet Yeah, I think so. Plus, @meeuw is trying to get all this to work with generators, in which case we don't want to loop through the whole output.

**Comment by meeuw** [comment]
hmm yes, I have to reconsider what to do, maybe refactor MyCli.output to include the auto vertical mode.

**Comment by meeuw** [comment]
Oh, I think this can actually be done with generators:
First, get the first_line: ```first_line = next(formatted)```
Next, chain the first_line back: ```formatted = itertools.chain([first_list], formatted)```

Is this ready for merging?

**Comment by tsroten** [comment]
@meeuw Let's check with @j-bennet, first. Irina, are you ok with just looking at the first line (re: your comment ^^)?

**Comment by j-bennet** [comment]
@tsroten @meeuw I think so. This is what we're doing right now anyway, so at least we're not breaking things.

One other thing we can consider though, since `cli_helpers` have to loop through all the lines, they know how long the longest was. Could we perhaps return this value from `format_output`, along with `rows` and `headers`?

**Comment by tsroten** [comment]
@j-bennet Good idea. CLI Helpers could return something like `rows, headers, metadata`, where `metadata` is a dict with various pertinent information (e.g. `rows`, `columns`, etc.)

That would have to be something for >= v1 of CLI Helpers.

**Comment by tsroten** [comment]
Merging for now. We can try to address CLI Helpers returning length in a later pull request.

**Pull request merged**